src/hotspot/share/gc/serial/serialHeap.cpp

```diff
@@ -147,15 +147,15 @@ GrowableArray<MemoryPool*> SerialHeap::memory_pools() {
   return memory_pools;
 }
 
-void SerialHeap::young_process_roots(OopClosure* root_closure,
-                                     OopIterateClosure* old_gen_closure,
-                                     CLDClosure* cld_closure) {
-  MarkingCodeBlobClosure mark_code_closure(root_closure, CodeBlobToOopClosure::FixRelocations, false /* keepalive nmethods */);
+void SerialHeap::young_process_roots(OopClosure* root_closure, CLDClosure* cld_closure) {
+  MarkingCodeBlobClosure mark_code_closure(root_closure,
+                                           CodeBlobToOopClosure::FixRelocations,
+                                           false /* keepalive nmethods */);
 
   process_roots(SO_ScavengeCodeCache, root_closure,
                 cld_closure, cld_closure, &mark_code_closure);
 
-  old_gen()->younger_refs_iterate(old_gen_closure);
+  old_gen()->process_old_to_young_refs();
 }
```

Comment:
The naming still needs some refinement. The word "process" is also a name that doesn't tell what operation it is performing. At the call to `process_roots` we pass in the provided closure, and that's the operation that is going to be applied. Whereas `process_old_to_young_refs` has the operation baked into the function, and should indicate what that operation is in the name.

This also shows that this functions becomes incoherent after this patch. One part operates on passed-in closures, while the second part uses a hard-coded closure. If you want to do this optimization, I think it would be better if you split `young_process_roots` into two separate functions. This will make it easier to follow what's going on in the code.